### PR TITLE
Add support for subcommands

### PR DESCRIPTION
**Pull request by matklad**
Hey, what do you think this API addition?

### CODE REVIEW

**Comment by matklad** [comment]
(I'd love to use it here: https://github.com/matklad/rust-analyzer/blob/91f9bc2b866fc87dbaadb29a8c72eba10ae57c5c/xtask/src/main.rs#L25-L26)

File: src/lib.rs [hunk 1]
```diff
@@ -152,6 +152,18 @@ impl Arguments {
         Arguments(args)
     }
 
+    /// Returns the name of the subcommand, that is, the first positional argument.
+    #[inline(never)]
```
Comment (matklad) [inline src/lib.rs]: Actually, does this inline do anything? I'd like to say something like "hey, even with LTO, don't bother inlining this", but I am not sure if lto actually looks at `inline(never)`?

Comment (RazrFalcon) [reply src/lib.rs]: According to the `cargo-bloat` it adds only 260B, so we can skip the `inline` attribute completely and let Rust decide what to do.

File: src/lib.rs [hunk 2]
```diff
@@ -152,6 +152,18 @@ impl Arguments {
...
+    pub fn subcommand(&mut self) -> Result<Option<String>, Error> {
+        if self.0.is_empty() {
+            return Ok(None);
+        }
+        self.0.remove(0)
```
Comment (RazrFalcon) [inline src/lib.rs]: Could you add a new line after `if {}`? Just for style.

**Comment by RazrFalcon** [comment]
I've never worked with subcommands, so I'm not sure how the api should like. If such simple change is fine for you - I will merge it.

**Comment by matklad** [comment]
> >If such simple change is fine for you - I will merge it.

It will definitely cover rust-analyer's use-case. It will even work for nested subcommands (like `rustup toolchain list`). There might be some nasty use-cases which I am not aware of though.

Pushed update to docs & test (and also handled an edge case).

**Comment by RazrFalcon** [comment]
Thanks! I will publish an update soon.

**Comment by matklad** [comment]
Thank you! And also thanks for building `pico-args` in the first place, it's the perfect crate for rust-analyzer use-case, where we use it for installation script (which every user compiles on their machine), so compile times are very important!

**Comment by RazrFalcon** [comment]
Thanks, I'm looking forward for `rust-analyzer` too.
